hazelcast-sql/src/main/java/com/hazelcast/sql/impl/expression/FieldAccessExpression.java

```diff
@@ -76,33 +78,29 @@ public T eval(Row row, ExpressionEvalContext context) {
     @Override
     public T eval(final Row row, final ExpressionEvalContext context, boolean useLazyDeserialization) {
         // Use lazy deserialization for nested queries. Only the last access should be eager.
-        final Object res = ref.eval(row, context, true);
-        if (res == null) {
+        final Object result = ref.eval(row, context, true);
+        if (result == null) {
             return null;
         }
 
-        if (isPrimitive(res.getClass())) {
+        if (isPrimitive(result.getClass())) {
             throw QueryException.error("Field Access expression can not be applied to primitive types");
         }
 
-        if (getterCache.get() == null) {
-            getterCache.compareAndSet(null, new EvictableGetterCache(
-                    MAX_CLASS_COUNT,
-                    MAX_GETTER_PER_CLASS_COUNT,
-                    GetterCache.EVICTABLE_CACHE_EVICTION_PERCENTAGE,
-                    false
-            ));
-        }
-
-        // defensive check, should never happen
-        final GetterCache cache = getterCache.get();
-        assert cache != null : "GetterCache should never be null";
+        getterCache.compareAndSet(null, new EvictableGetterCache(
```

Comment:
you create `EvictableGetterCache` in each eval invocation, second parameter is not a Supplier!